selfdrive/ui/tests/test_translations.py

```diff
@@ -5,23 +5,32 @@
 import shutil
 import unittest
 import xml.etree.ElementTree as ET
+from parameterized import parameterized_class
 
 from openpilot.selfdrive.ui.update_translations import TRANSLATIONS_DIR, LANGUAGES_FILE, update_translations
 
+with open(LANGUAGES_FILE, "r") as f:
+  translation_files = json.load(f)
+
+translations_updated = False
+
 TMP_TRANSLATIONS_DIR = os.path.join(TRANSLATIONS_DIR, "tmp")
 UNFINISHED_TRANSLATION_TAG = "<translation type=\"unfinished\""  # non-empty translations can be marked unfinished
 LOCATION_TAG = "<location "
 FORMAT_ARG = re.compile("%[0-9]+")
 
 
+@parameterized_class(("name", "file"), translation_files.items())
 class TestTranslations(unittest.TestCase):
   @classmethod
   def setUpClass(cls):
-    with open(LANGUAGES_FILE, "r") as f:
-      cls.translation_files = json.load(f)
-
+    super(TestTranslations, cls).setUpClass()
+    global translations_updated
     # Set up temp directory
     shutil.copytree(TRANSLATIONS_DIR, TMP_TRANSLATIONS_DIR, dirs_exist_ok=True)
+    if not translations_updated:
+      update_translations(plural_only=["main_en"], translations_dir=TMP_TRANSLATIONS_DIR)
+      translations_updated = True
```

Comment:
this is not safe at all. just avoid using the global entirely by making a tmp dir per-class using tempfile.NamedTemporaryDirectory